src/components/TodoList.tsx

```diff
@@ -0,0 +1,82 @@
+import { Dispatch, SetStateAction } from 'react';
+import { TransitionGroup, CSSTransition } from 'react-transition-group';
+import { TodoItem } from './TodoItem';
+import { ErrorType, Todo } from '../types';
+import { deleteTodo } from '../api/todos';
+
+interface Props {
+  todos: Todo[],
+  tempTodo: Todo | null,
+  setTodos: Dispatch<SetStateAction<Todo[]>>
+  isLoading: number | boolean
+  setIsLoading: Dispatch<SetStateAction<number | boolean>>
+  handleError: (error: ErrorType) => void
+}
+
+export const TodoList: React.FC<Props> = (props) => {
+  const {
+    todos,
+    tempTodo,
+    setTodos,
+    isLoading,
+    setIsLoading,
+    handleError,
+  } = props;
+
+  const handleTodoDelete = (id: number) => {
+    setIsLoading(id);
+
+    const deletingTodo = async () => {
+      try {
+        if (todos) {
+          await deleteTodo(id);
+          setTodos(prev => prev.filter(todo => todo.id !== id));
+        }
+      } catch (error) {
+        handleError(ErrorType.DELETE);
+      } finally {
+        setIsLoading(false);
```

Comment:
Now only one item can be in "loading" state. Try to do it in array with ID items that are loading instead of just boolean